internal/instance/resource_instance_test.go

```diff
@@ -216,6 +216,55 @@ func TestAccInstance_restartVirtualMachine(t *testing.T) {
 	})
 }
 
+func TestAccInstance_renameInstance(t *testing.T) {
+	instanceNameA := acctest.GenerateName(3, "-")
+	instanceNameB := acctest.GenerateName(3, "-")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				// Launch a new instance.
+				Config: testAccInstance_rename(instanceNameA, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameA),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Running"),
+				),
+			},
+			{
+				// Stop and rename the instance.
+				Config: testAccInstance_rename(instanceNameB, false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameB),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Stopped"),
+				),
+			},
+			{
+				// Rename the instance while stopped.
+				Config: testAccInstance_rename(instanceNameA, false),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameA),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Stopped"),
+				),
+			},
+			{
+				// Rename and start the instance.
+				Config: testAccInstance_rename(instanceNameB, true),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameB),
+					resource.TestCheckResourceAttr("lxd_instance.instance1", "status", "Running"),
+				),
+			},
+			{
+				// Ensure instance rename fails when instance is running.
+				Config:      testAccInstance_rename(instanceNameA, true),
+				ExpectError: regexp.MustCompile("Renaming of running instance not allowed"),
```

Comment:
Should we also have the `resource.TestCheckResourceAttr("lxd_instance.instance1", "name", instanceNameB),` checked here?

Reply:
Not sure if state is testable when error is expected, but worth trying.

Reply:
Seems to be working. Thanks :)